src/main/java/fruitshop/strategy/PurchaseOperation.java

```diff
@@ -0,0 +1,20 @@
+package fruitshop.strategy;
+
+import fruitshop.db.Storage;
+import fruitshop.model.FruitTransaction;
+
+public class PurchaseOperation implements OperationHandler {
+    private final Storage storage = new Storage();
+
+    @Override
+    public void handler(FruitTransaction fruitTransaction) {
+        int newQuantity = storage.getQuantity(fruitTransaction.getFruit())
+                - fruitTransaction.getQuantity();
+        if (newQuantity >= 0) {
+            storage.put(fruitTransaction.getFruit(), newQuantity);
+        } else {
+            throw new RuntimeException("negative balance cannot be recorded at "
+                    + PurchaseOperation.class);
```

Comment:
The exception message should provide more details about the error. Include the fruit name and the attempted quantity that led to the negative balance.